### PR TITLE
feat: update and deploy oov2 subgraph

### PR DESCRIPTION
**Pull request by ryanwolhuter**
Signed-off-by: ryanwolhuter <dev@ryanwolhuter.com>

### Motivation

Add a new subgraph that accurately represents the OOV2. 

OOV2 allows some new functions `setBond`, `setCustomLiveness` and `setEventBased`. These are not present in events and therefore must come from function call handlers instead. The other events for OOV1 and OOV2 are the same, so the mappings are unchanged.

### Changes

* add new ABI
* update contract addresses and start blocks in manifest
*  add `bond`, `customLiveness` and `eventBased` fields to schema
* add function call handlers for `setBond`, `setCustomLiveness` and `setEventBased`

### CODE REVIEW

**Review by daywiss — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -24,6 +24,18 @@ This subgraph indexes events emitted by the "Optimistic Oracle" contracts. The c
...
+## Optimistic Oracle V2 Events and Calls:
+
+This subgraph indexes events and function calls by the "Optimistic Oracle V2" contracts. The code can be found in `packages/optimistic-oracle-v2`
+
+- (unofficial) Mainnet: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/mainnet-oo-v2-staging
+- (unofficial) Polygon: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/polygon-oo-v2-staging
+- (unofficial) Arbitrum: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/arbitrum-oo-v2-staging
+- (unofficial) Optimism: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/optimism-oo-v2-staging
+- (unofficial) Boba: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/boba-oo-v2-staging
+- (staging) Goerli: https://thegraph.com/hosted-service/subgraph/ryanwolhuter/goerli-oo-v2-staging
```
Comment (md0x) [inline README.md:36]: Are they deployed?
The links doesn't work

File: packages/optimistic-oracle-v2/README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Optimistic Oracle V2 Subgraph
+
+## Deploying the subgraph
+
+- Mainnet: `yarn prepare:mainnet && yarn deploy:mainnet`
+- Kovan: `yarn prepare:kovan && yarn deploy:kovan`
```
Comment (md0x) [inline packages/optimistic-oracle-v2/README.md:6]: Kovan can be removed and maybe add the [rest of networks](https://github.com/UMAprotocol/subgraphs/blob/master/packages/optimistic-asserter/README.md)

File: packages/optimistic-oracle-v2/manifest/templates/OptimisticOracleV2.template.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    eventHandlers:
+      - event: RequestPrice(indexed address,bytes32,uint256,bytes,address,uint256,uint256)
+        handler: handleOptimisticRequestPrice
+      - event: ProposePrice(indexed address,indexed address,bytes32,uint256,bytes,int256,uint256,address)
+        handler: handleOptimisticProposePrice
+      - event: DisputePrice(indexed address,indexed address,indexed address,bytes32,uint256,bytes,int256)
+        handler: handleOptimisticDisputePrice
+      - event: Settle(indexed address,indexed address,indexed address,bytes32,uint256,bytes,int256,uint256)
+        handler: handleOptimisticSettle
+    callHandlers:
```
Comment (md0x) [inline packages/optimistic-oracle-v2/manifest/templates/OptimisticOracleV2.template.yaml:28]: Not sure if polygon accepts this kind of handlers, we need to check this when deploying

File: README.md (hunk 1 above)
Comment (ryanwolhuter) [reply README.md]: whoops I made a mistake in the urls lol, please take another look. they are all deployed

File: packages/optimistic-oracle-v2/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export {
+  handleOptimisticRequestPrice,
+  handleOptimisticProposePrice,
+  handleOptimisticDisputePrice,
+  handleOptimisticSettle
+} from "./mappings/optimisticOracleV2";
```
[the remark this answers is not in the document]
Comment (ryanwolhuter) [reply packages/optimistic-oracle-v2/src/index.ts]: added 👍

**Review by md0x — COMMENTED** [review]
Polygon subgraph is failing because it doesn't support `callHandlers`.

We need to handle differently the eventBased and customLiveness checks in Polygon.

We can remove the callHandlers section in the subgraph.yml file when doing `yarn prepare:polygon` and then in the subgraph mapping:
```
import { dataSource } from "@graphprotocol/graph-ts";

let network = dataSource.network();

let isPolygon = network === "matic";

if(isPolygon){
// Fetch request in oov2 with getRequest and check if its an event based price request and the
// customLiveness...
}
```

**Review by md0x — APPROVED** [review]
LGTM!
Thanks for the good work and the latest Polygon improvement!

**Review by chrismaree — APPROVED** [review]

**Pull request merged**